memory/src/lib.rs

```diff
@@ -190,15 +190,32 @@ impl MemoryChip {
     }
 
     fn insert_dummy_reads(ops: &mut Vec<(u32, Operation)>) {
+        if ops.is_empty() {
+            return;
+        }
+
         let table_len = ops.len() as u32;
         let mut dummy_ops = Vec::new();
         for (op1, op2) in ops.iter().zip(ops.iter().skip(1)) {
             let addr_diff = op2.1.get_address() - op1.1.get_address();
             if addr_diff != 0 {
-                // TODO: We should add dummy reads when addr_diff is greater than
-                // the number of operations
-                continue;
+                // Add dummy reads when addr_diff is greater than the number of operations
+                if addr_diff > table_len {
+                    let num_dummy_ops = addr_diff / table_len;
```

Comment:
Hm this would have pretty bad worst-case behavior if there was a tiny execution that touched 0 and then 2^28 or something. (We'd calculate something close to 2^28 rows, when we only need the sqrt of that.) Maybe that would never happen depending on the allocator. Probably okay for now but should improve at some point.

Reply:
Will revisit after we settle on a solution for #40